cdap-watchdog/src/main/java/co/cask/cdap/logging/plugins/LocationTriggeringPolicyBase.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.logging.plugins;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.LogbackException;
+import ch.qos.logback.core.spi.ContextAwareBase;
+import org.apache.twill.filesystem.Location;
+
+import java.io.File;
+
+/**
+ * Location triggering policy base class
+ */
+public abstract class LocationTriggeringPolicyBase extends ContextAwareBase implements LocationTriggeringPolicy {
+  protected Location activeLocation;
+  protected long activeLocationSize;
```

Comment:
terence suggested to avoid protected usage and he suggested alternate approach right? can you make that change.